sdk/message/src/legacy.rs

```diff
@@ -5,30 +5,58 @@
 //! `v0` is a [future message format] that encodes more account keys into a
 //! transaction than the legacy format.
 //!
-//! [`legacy`]: crate::message::legacy
-//! [`v0`]: crate::message::v0
+//! [`legacy`]: crate::legacy
+//! [`v0`]: crate::v0
 //! [future message format]: https://docs.solanalabs.com/proposals/versioned-transactions
 
 #![allow(clippy::arithmetic_side_effects)]
 
-#[cfg(target_arch = "wasm32")]
-use crate::wasm_bindgen;
 #[allow(deprecated)]
 pub use builtins::{BUILTIN_PROGRAMS_KEYS, MAYBE_BUILTIN_KEY_OR_SYSVAR};
+#[cfg(feature = "serde")]
+use serde_derive::{Deserialize, Serialize};
+#[cfg(feature = "frozen-abi")]
+use solana_frozen_abi_macro::{frozen_abi, AbiExample};
+#[cfg(target_arch = "wasm32")]
+use wasm_bindgen::prelude::wasm_bindgen;
 use {
     crate::{
-        bpf_loader, bpf_loader_deprecated, bpf_loader_upgradeable,
-        hash::Hash,
-        instruction::{CompiledInstruction, Instruction},
-        message::{compiled_keys::CompiledKeys, MessageHeader},
-        pubkey::Pubkey,
-        system_instruction, system_program, sysvar,
+        compiled_instruction::CompiledInstruction, compiled_keys::CompiledKeys, MessageHeader,
     },
+    solana_hash::Hash,
+    solana_instruction::Instruction,
+    solana_pubkey::Pubkey,
     solana_sanitize::{Sanitize, SanitizeError},
-    solana_short_vec as short_vec,
+    solana_sdk_ids::{
+        bpf_loader, bpf_loader_deprecated, bpf_loader_upgradeable, system_program, sysvar,
+    },
     std::{collections::HashSet, convert::TryFrom, str::FromStr},
 };
 
+// copied from deprecated code in solana_program::sysvar to avoid a dependency.
+// This should be removed when the items that depend on it are removed.
+lazy_static::lazy_static! {
+    // This will be deprecated and so this list shouldn't be modified
+    static ref ALL_IDS: Vec<Pubkey> = vec![
+        sysvar::clock::id(),
+        sysvar::epoch_schedule::id(),
+        sysvar::fees::id(),
+        sysvar::recent_blockhashes::id(),
+        sysvar::rent::id(),
+        sysvar::rewards::id(),
+        sysvar::slot_hashes::id(),
+        sysvar::slot_history::id(),
+        sysvar::stake_history::id(),
+        sysvar::instructions::id(),
+    ];
+}
```

Comment:
This makes sense to me. To be safe, can you add a test to make sure that `ALL_IDS == solana_sysvar::ALL_IDS`?